### ISSUE TITLE
VC-1 playback broken on AMD APU

### ISSUE THREAD

**Issue by Psych05** [OpenELEC/OpenELEC.tv#4423]
THE PROBLEM
I own a Zotac Zbox Nano AQ01 (AMD Kabini). Whenever I start any video file encoded with the VC-1 codec, it basically results in a screen completely of artifacts. This did not happen in any of the 5.x.x versions, but was already present in 6.0 beta 5 (i didn't try the earlier betas).
As two people have confirmed the issue in my forum post (http://openelec.tv/forum/113-video-decoders/78533-vc-1-unwatchable-in-6-0-0), i'm dragging this here. I hope this is the correct way.

THE LOG FILE
xbmclogs.com/pq61wymhd (component "Video" enabled)
What happens in the log:
 - I start a VC-1 encoded video file, let it play for a few seconds
 - The resulting screen output consists almost only of artifacts and flickering, completely unwatchable
 - I stop the playback and start kodi log uploader

ADDITIONAL INFORMATION
For the log file, standard settings were used in Video -> Acceleration. I also tried other settings there. But no matter what I choose, VDPAU decoding is always used, except if i disable it for vc-1 or globally. Then, software decoding is used, resulting in heavy stuttering. All the other settings on the same page don't solve the problem, i tried them all.

**Comment by fritsch** [comment]
Thanks for your report. That's very likely, sadly - the kodi development team currently does not run any AMD hardware anymore.

Could you do me a favor and test a @Milhouse Generic build from the forum, here: http://forum.kodi.tv/showthread.php?tid=238393

**Comment by Psych05** [comment]
It really concerns me to hear that, as i was planning to use this as my HTPC system for years to come. It has proven very reliable and so far did handled i wanted with ease. But your statement doesn't leave too much hope for future support, does it?

Back on topic, i tried the latest build from the link you provided (#1103). As far as i can tell, the behavior regarding VC-1 looks exactly the same.

**Comment by fritsch** [comment]
Yeah - that's sad. So it's a regression in mesa which needs filing on bugs.freedesktop.org - I sadly cannot file it as I don't have this kind of hardware.

**Comment by fritsch** [comment]
http://cgit.freedesktop.org/mesa/mesa/commit/?h=11.0&id=6a958b0b51a79bf4f79a898db5c0277076e138d9 <- see - there is hope, even for simple / main

**Comment by MilhouseVH** [comment]
Should be fixed in latest test build, 1121:

http://forum.kodi.tv/showthread.php?tid=238393&pid=2167501#pid2167501

**Comment by chewitt** [comment]
@Psych05 - please confirm this is fixed in a current milhouse build. Thanks.

**Comment by Psych05** [comment]
Happy to confirm, fixed in 1121. 
In which timeframe will this realistically make it to a stable version? Can't wait to get back on board!

**Comment by fritsch** [comment]
1121 runs v17 code ... but for your fix mesa 11.0.x is the culprit which had the fix I linked above.

**Comment by Psych05** [comment]
I don't know - i really wish i had better news. But the behavior regarding VC-1 is absolutely unchanged. Same in build 1128.

**Comment by fritsch** [comment]
Good - then nothing we can do for you.

**Comment by Klojum** [comment]
FYI: Just tested a VC-1 video on OpenELEC 6.0.1 (not yet released), on a AMD A4-5300 apu (Radeon HD 7480D) machine here. No problems with artifacts or other glitches:  https://i.imgur.com/PvbleJa.jpg

**Comment by Psych05** [comment]
And there is hope again, thank you for the good news! You know, kodibuntu is really not for me. I can't wait to get back on board.
Can this version be downloaded somewhere for me to test?

**Comment by Klojum** [comment]
@Psych05 : Sorry, testing is still ongoing for other parts of OE as well. Patience, my friend  :-)

**Comment by Klojum** [comment]
@Psych05 : I've been told there is very similar test build you can try:  :-)
http://snapshots.openelec.tv/OpenELEC-Generic.x86_64-6.0-devel-20160104091551-r21696-gac9f6fb.tar
FYI, I got my VC-1 test video from here:
http://download.openbricks.org/sample/VC1/

**Comment by Psych05** [comment]
Testet the build you linked with no success. Also tried the video sample file you provided, as well as a few others from my NAS. All with the usual glitches.
This is very strange, as mine is an A4-5000, one would think that this is 99% the same APU. I also checked that the same decoder is used (ff-vc1-vdpau), which is the case. Nothing more I can do here at the moment, i guess.

**Comment by Klojum** [comment]
Your GPU is a 8000 series, mine is a 7000 series, the only difference I can think of. And yes, it's annoying to say the least.  You did disable the VAAPI acceleration, I take it?  Maybe the GPU memory allocation in the BIOS is insufficient?
